File: Practice/NCodeAlgoDS/DynamicArrays/dynamic_array.py
class Array:
    def __init__(self):
        self.capacity = 2
        self.length = 0
        self.arr = [0] * 2
    
    # Insert n in the last position of the array
    def pushback(self, n):
        if self.length == self.capacity:
            self.resize()
        # insert at next empty position 
        self.arr[self.length] = n
        self.length += 1

    def resize(self):
        # Create new array of double capacity
        self.capacity = 2 * self.capacity
        newArr = [0] * self.capacity

        # Copy elements to newArr
        for i in range(self.length):
            newArr[i] = self.arr[i]
        
        self.arr = newArr
    
    # Remove the last element in the array
    def popback(self):
        if self.length > 0:
            self.length -= 1
        return -1
    
    # Get value at i-th index
    def get(self, i):
        if i < self.length:
            return self.arr[i]
        return -1 # out of bounds or throw exception
    
    # Insert n at i-th index
    def insert(self, i, n):
        if i < self.length:
            self.arr[i] = n
            return
    
    def print(self):
        for i in range(self.length):
            print(self.arr[i])
        print()

    

File: Practice/NCodeAlgoDS/DynamicArrays/dynamic_array.swift
// Swift arrays are dynamic when using 'var' to declare the array, but this is an example of resizing
class Array {
    var capacity = 2
    var length = 0
    var arr = Swift.Array(repeating: 0, count: 2) // Array of capacity 2

    func pushback(_ n: Int) {

        if self.length == self.capacity {
            resize()
        }
        self.arr[length] = n
        self.length += 1
    }

    func resize() {
        // Create a new array of double capacity
        var newArray = Swift.Array(repeating: 0, count: self.capacity * 2) 
        
        // Copy elements to newArr
        for i in 0..<self.length {
            newArray[i] = self.arr[i]
        }
        self.arr = newArray
    }

    // Remove the last element in the array
    func popback() {
        if self.length > 0 {
            self.length -= 1
        }
    }

    // Get value at i-th index
    func get(_ i: Int) -> Int {
        if i < self.length {
            return self.arr[i]
        }
        return -1 // Out of bounds
    }

    func print() {
        for i in 0..<self.length {
            Swift.print(arr[i], terminator: " ")
        }
        Swift.print()
    }
}